india_compliance/gst_india/doctype/gst_hsn_code/gst_hsn_code.py

```diff
@@ -23,28 +23,91 @@ def update_taxes_in_item_master(taxes, hsn_code):
     return 1
 
 
-@enable_autocommit
+BATCH_SIZE = 100
+
+
 def update_item_document(taxes, hsn_code):
+    """Update taxes for all items with the given HSN code using bulk operations for performance."""
     taxes = frappe.parse_json(taxes)
-    items = frappe.get_list("Item", filters={"gst_hsn_code": hsn_code})
+    items = frappe.get_list("Item", filters={"gst_hsn_code": hsn_code}, pluck="name")
+
+    if not items:
+        return
+
+    for i in range(0, len(items), BATCH_SIZE):
+        batch_items = items[i : i + BATCH_SIZE]
+        _process_item_batch(batch_items, taxes)
+
+
+def _process_item_batch(batch_items, taxes):
+    frappe.db.delete("Item Tax", {"parent": ["in", batch_items]})
+
+    if taxes:
+        _bulk_insert_item_taxes(batch_items, taxes)
 
-    for index, item in enumerate(items):
-        item_to_be_updated = frappe.get_doc("Item", item.name)
-        item_to_be_updated.taxes = []
-        for tax in taxes:
+    _update_item_modified_timestamp(batch_items)
+
+    # TODO: Add Versioning?
```

Comment:
Only if track changes are enabled, constant text in version, saying item taxes updated from HSN